src/main/java/p455w0rd/danknull/inventory/cap/DankNullCapabilityProvider.java

```diff
@@ -26,13 +28,31 @@ public DankNullCapabilityProvider(final ModGlobals.DankNullTier tier, final Item
 			@Override
 			protected void onContentsChanged(final int slot) {
 				super.onContentsChanged(slot);
-				stack.setTagCompound((NBTTagCompound) CapabilityDankNull.DANK_NULL_CAPABILITY.writeNBT(this, null));
+				NBTTagCompound oldNBT = stack.getTagCompound();
+				NBTTagCompound newNBT = (NBTTagCompound) CapabilityDankNull.DANK_NULL_CAPABILITY.writeNBT(this, null);
+				if (Objects.nonNull(oldNBT)) {
+					if (Objects.nonNull(newNBT)) {
+						oldNBT.merge(newNBT);
+					}
+				} else {
+					oldNBT = newNBT;
+				}
+				stack.setTagCompound(oldNBT);
 			}
 
 			@Override
 			protected void onSettingsChanged() {
 				super.onSettingsChanged();
-				stack.setTagCompound((NBTTagCompound) CapabilityDankNull.DANK_NULL_CAPABILITY.writeNBT(this, null));
+				NBTTagCompound oldNBT = stack.getTagCompound();
+				NBTTagCompound newNBT = (NBTTagCompound) CapabilityDankNull.DANK_NULL_CAPABILITY.writeNBT(this, null);
+				if (Objects.nonNull(oldNBT)) {
```

Comment:
Rather stay will non OO style for null checks. Keep the project consistent. A full conversion to function based checks could be done in a single VCS.